test/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,564 @@
+			Assert.fail("Unexpected DumpConfigurationUnavailableException : " + e.getMessage());
+		}
+
+		List<String> newDumpOptions = new ArrayList<String>();
+		newDumpOptions = getDumpOptions();
+		boolean res = initialDumpOptions.equals(newDumpOptions);
+		
+		Assert.assertTrue(res, "Reset Dump Options failed");
+		
+	}
+
+	
+	/**
+	 * Function to check if the specified file is present in the directory path 
+	 *
+	 * @param dirPath - Directory path where dumps are created 
+	 * @param fileName - Dump file name
+	 * @return a boolean indicating if the file is found or not
+	 */	
+	private static boolean findAndDeleteFile(String dirPath, String fileName) {
+		BufferedReader reader = null;
+		boolean found = false;
+		File dir = new File(dirPath);
+
+        	File[] files = dir.listFiles();
+
+        	for (File f: files) {
+			found = f.getName().startsWith(fileName);
+        	        if (found) {
+        	      		try {
+                    			reader = new BufferedReader(new FileReader(f));
+					reader.close();
+					boolean res = f.delete();	
+					Assert.assertTrue(res, "Failed to delete the file");
+						
+	                   		break;
+	        	        } catch (java.io.FileNotFoundException e) {
+					Assert.fail("Unexpected FileNotFoundException : " + e.getMessage());
+	        	        } catch (java.io.IOException e) {
+					Assert.fail("Unexpected IOException : " + e.getMessage());
+				}
+        		}
+        	}
+
+		if (null == reader) {
+			found = false;
+		} else {
+			found = true;
+		}
+
+
+		return found;
+	}
+
+
+	/**
+	 * Internal function: Starts a remote server to connect to.
+	 */
+	private static void startRemoteServer() {
+		logger.info("Start Remote Server!");
+		ArrayList<String> argBuffer = new ArrayList<String>();
+		char fs = File.separatorChar;
+		String javaExec = System.getProperty("java.home") + fs + "bin" + fs + "java";
+		argBuffer.add(javaExec);
+		argBuffer.add("-Xdump:dynamic");
+		argBuffer.add("-classpath");
+		argBuffer.add(System.getProperty("java.class.path"));
+		argBuffer.add("-Dcom.sun.management.jmxremote.port=9999");
+		argBuffer.add("-Dcom.sun.management.jmxremote.authenticate=false");
+		argBuffer.add("-Dcom.sun.management.jmxremote.ssl=false");
+		argBuffer.add(RemoteProcess.class.getName());
+		String cmdLine[] = new String[argBuffer.size()];
+		argBuffer.toArray(cmdLine);
+		String cmds = "";
+		for (int i = 0; i < cmdLine.length; i++) {
+			cmds = cmds + cmdLine[i] + " ";
+		}
+
+		logger.info(cmds);
+		try {
+			remoteServer = Runtime.getRuntime().exec(cmds);
+		} catch (IOException e) {
+			e.printStackTrace();
+			Assert.fail("Failed to launch child process " + remoteServer);
+		}
+
+		try {
+			Thread.sleep(10);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+	} 
+
+	/**
+	 * Internal function: Stops a remote server.
+	 */
+	private static void stopRemoteServer() {
+		watchdog.interrupt();
+		remoteServer.destroy();
+		try {
+			Thread.sleep(10000);
```

Comment:
Must the test be delayed a full 10 seconds? Cannot we detect termination of the process without such a long wait?